### DISCUSSION TITLE
using the "*" operator for Cartesian product of two collections.

### DISCUSSION THREAD

**Discussion by VardanKosakyan** [dotnet/csharplang#6788] [Language Ideas]
Currently operator overloading is not supported in extension methods. I believe it will be supported in the future. It is proposed to implement overloaded "*" operator extension for `IEnumerable<T>`, which will realize Cartesian product of  two collections.  Result of Cartesian product of two collections `IEnumerable<T1>` and `IEnumerable<T2>` will be collection of tuples `IEnumerable<(T1,T2)>`.
It will have more beautiful and readable syntax.

example
```
var col1 = new List<string> { "one", "two", "three"};
var col2 = new List<int> { 1, 2, 3, 4 };

//proposed syntax
foreach( (text, number)  in col1 * col2)
{
        Console.Write(" number {number} text {text} ";
}

//alternative
foreach( (text, number)  in col1.SelectMany( col2, (t, n) => (t, n) ))
{
        Console.Write(" number {number} text {text} ";
}

//or
foreach( text in col1)
{
       foreach( number in col2)
       {
                  Console.Write(" number {number} text {text} ";
       }
}

```

**Comment by HaloFour** [comment]
IMO this is a better use case for an extension method that makes it clear that the result is a Cartesian product.  Different people will have different expectations as to what multiplying two collections should mean.
